libs/cart/driver/magento/src/cart-item.service.ts

```diff
@@ -115,42 +109,26 @@ export class DaffMagentoCartItemService implements DaffCartItemServiceInterface
   }
 
   private addBundledProduct(cartId: DaffCart['id'], cartItemInput: DaffCompositeCartItemInput): Observable<Partial<DaffCart>> {
-    const bundleInput = transformCompositeCartItem(cartItemInput);
-    return this.mutationQueue.mutate<MagentoAddBundleCartItemResponse>({
-      mutation: addBundleCartItem(this.extraCartFragments),
-      variables: {
-        cartId,
-        input: bundleInput.input,
-        options: bundleInput.options,
-      },
-    }).pipe(
-      map(result => this.cartTransformer.transform(result.data.addBundleProductsToCart.cart)),
-    );
+    return this.addCartItemRequest(cartId, transformCompositeCartItem(cartItemInput));
   }
 
   private addConfigurableProduct(cartId: DaffCart['id'], cartItemInput: DaffConfigurableCartItemInput): Observable<Partial<DaffCart>> {
-    const configurableInput: MagentoConfigurableCartItemInput = transformConfigurableCartItem(cartItemInput);
-    return this.mutationQueue.mutate<MagentoAddConfigurableCartItemResponse>({
-      mutation: addConfigurableCartItem(this.extraCartFragments),
-      variables: {
-        cartId,
-        parentSku: configurableInput.parentSku,
-        data: configurableInput.data,
-      },
-    }).pipe(
-      map(result => this.cartTransformer.transform(result.data.addConfigurableProductsToCart.cart)),
-    );
+    return this.addCartItemRequest(cartId, transformConfigurableCartItem(cartItemInput));
   }
 
   private addSimpleProduct(cartId: DaffCart['id'], cartItemInput: DaffCartItemInput): Observable<Partial<DaffCart>> {
-    return this.mutationQueue.mutate<MagentoAddSimpleCartItemResponse>({
-      mutation: addSimpleCartItem(this.extraCartFragments),
+    return this.addCartItemRequest(cartId, transformSimpleCartItem(cartItemInput));
```

Comment:
The magento docs for (https://devdocs.magento.com/guides/v2.4/graphql/mutations/add-products-to-cart.html#input-attributes) say that all of the cart item inputs are required, which I admit sounds weird. But if it is the case, then adding a simple product to the cart like this will fail I think. Either `transformSimpleCartItem` needs to return something for all of the input fields or default values need to be added for the graphQl queries somehow.

Reply:
I think the docs say only `sku` and `quantity` are required. In any case the following input works through postman:
```json
"input": {
    "quantity": 1,
    "sku": "Tee-Brush-Spring Blister"
}
```

Reply:
Note that `[ID!]` is different from `[ID!]!`

Reply:
I guess I took "The CartItemInput object must contain the following attributes:" too literally.